src/wazuh_db/wdb_global.c

```diff
@@ -791,38 +791,41 @@ int wdb_global_insert_agent_belong(wdb_t *wdb, int id_group, int id_agent, int p
     }
 }
 
-int wdb_global_delete_group_belong(wdb_t *wdb, char* group_name) {
-    sqlite3_stmt *stmt = NULL;
-
-    if (!wdb->transaction && wdb_begin2(wdb) < 0) {
-        mdebug1("Cannot begin transaction");
-        return OS_INVALID;
-    }
-
-    if (wdb_stmt_cache(wdb, WDB_STMT_GLOBAL_DELETE_GROUP_BELONG) < 0) {
-        mdebug1("Cannot cache statement");
+int wdb_global_remove_agent_belong(wdb_t *wdb, int id_group, int id_agent) {
+    sqlite3_stmt *stmt = wdb_init_stmt_in_cache(wdb, WDB_STMT_GLOBAL_DELETE_AGENT_BELONG);
+    if (stmt == NULL) {
         return OS_INVALID;
     }
+    sqlite3_bind_int(stmt, 1, id_group);
+    sqlite3_bind_int(stmt, 2, id_agent);
 
-    stmt = wdb->stmt[WDB_STMT_GLOBAL_DELETE_GROUP_BELONG];
+    return wdb_exec_stmt_silent(stmt);
+}
 
-    if (sqlite3_bind_text(stmt, 1, group_name, -1, NULL) != SQLITE_OK) {
-        merror("DB(%s) sqlite3_bind_text(): %s", wdb->id, sqlite3_errmsg(wdb->db));
-        return OS_INVALID;
+bool wdb_is_group_empty(wdb_t *wdb, char* group_name) {
+    sqlite3_stmt* stmt = wdb_init_stmt_in_cache(wdb, WDB_STMT_GLOBAL_GROUP_BELONG_FIND);
+    if (stmt == NULL) {
+        return false;
     }
 
+    sqlite3_bind_text(stmt, 1, group_name, -1, NULL);
     switch (wdb_step(stmt)) {
-    case SQLITE_ROW:
     case SQLITE_DONE:
-        return OS_SUCCESS;
-        break;
+        return true;
+    case SQLITE_ROW:
+        return false;
     default:
-        mdebug1("SQLite: %s", sqlite3_errmsg(wdb->db));
-        return OS_INVALID;
+        mdebug1("SQL statement execution failed");
+        return false;
     }
 }
 
 int wdb_global_delete_group(wdb_t *wdb, char* group_name) {
+    if (!wdb_is_group_empty(wdb, group_name)) {
+        mdebug1("Unable to delete group '%s', an agent is still belonging to it", group_name);
```

Comment:
Maybe we could rephrase `an agent is still belonging to it`. Something like ` the group still contains agents`, the `group isn't empty` or similar.

Reply:
Fixed!